qiskit/circuit/quantumcircuit.py

```diff
@@ -823,6 +823,13 @@ def compose(
         else:
             dest = self.copy()
 
+        # If self does not have any cregs but other does then we allow
+        # that and add the registers to the output dest
+        if isinstance(other, QuantumCircuit):
+            if not self.clbits and other.clbits:
+                for reg in other.cregs:
+                    dest.add_register(reg)
```

Comment:
```suggestion
        # If self does not have any clbits but other does then we allow
        # that and add them to the output dest
        if isinstance(other, QuantumCircuit):
            if not self.clbits and other.clbits:
                dest.add_bits(other.clbits)
                for reg in other.cregs:
                    dest.add_register(reg)
```
just to catch any registerless bits, since registers aren't required any more

Reply:
Can you give me an example?  I have no idea what a "registerless bit" is.

Reply:
Ahh, never knew that.  Will add the test.  Many thanks.